ios/Runner/AppDelegate.swift

```diff
@@ -37,4 +45,26 @@ let apnTokenKey = "apnToken"
         // Save the token to use in the future
         UserDefaults.standard.set(token, forKey: apnTokenKey)
     }
+    
+    override func userNotificationCenter(_ center: UNUserNotificationCenter,
+                                willPresent notification: UNNotification,
+      withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        let userInfo = notification.request.content.userInfo
+        twakeApnChannel?.invokeMethod("willPresent", arguments: userInfo)
+        switch UIApplication.shared.applicationState {
+        case .active:
+            // Do not show notification when app in foreground
```

Comment:
why not to put back to `dart` logic to handle the same logic with android? How about the notification when you stay in chat list, in other chat?

Reply:
IMO: In foreground still shows notification and pls handle in this case

Reply:
done